components/raftstore-v2/src/raft/peer.rs

```diff
@@ -1,24 +1,52 @@
 // Copyright 2022 TiKV Project Authors. Licensed under Apache-2.0.
 
-use std::{collections::VecDeque, mem, sync::Arc};
+use std::{mem, sync::Arc};
 
+use crossbeam::atomic::AtomicCell;
 use engine_traits::{KvEngine, OpenOptions, RaftEngine, TabletFactory};
-use kvproto::{metapb, raft_serverpb::RegionLocalState};
-use raft::{RawNode, StateRole, INVALID_ID};
-use raftstore::store::{util::find_peer, Config, EntryStorage, RaftlogFetchTask, WriteRouter};
-use slog::{o, Logger};
-use tikv_util::{box_err, config::ReadableSize, worker::Scheduler};
+use fail::fail_point;
+use kvproto::{
+    metapb,
+    raft_cmdpb::{self, RaftCmdRequest},
+};
+use protobuf::Message;
+use raft::{RawNode, StateRole};
+use raftstore::{
+    store::{
+        fsm::Proposal,
+        metrics::PEER_PROPOSE_LOG_SIZE_HISTOGRAM,
+        peer::{
+            get_sync_log_from_request, CmdEpochChecker, ProposalContext, ProposalQueue,
+            RequestInspector,
+        },
+        util::{Lease, RegionReadProgress},
+        Config, EntryStorage, RaftlogFetchTask, ReadIndexQueue, ReadIndexRequest, Transport,
+    },
+    Error,
+};
+use slog::{debug, error, info, o, warn, Logger};
+use tikv_util::{
+    box_err,
+    config::ReadableSize,
+    time::{monotonic_raw_now, Instant as TiInstant},
+    worker::Scheduler,
+    Either,
+};
 
 use super::storage::Storage;
 use crate::{
+    batch::StoreContext,
     operation::AsyncWriter,
+    router::{CmdResChannel, QueryResChannel},
     tablet::{self, CachedTablet},
     Result,
 };
 
+const REGION_READ_PROGRESS_CAP: usize = 128;
+
 /// A peer that delegates commands between state machine and raft.
 pub struct Peer<EK: KvEngine, ER: RaftEngine> {
-    raft_group: RawNode<Storage<ER>>,
+    pub(crate) raft_group: RawNode<Storage<ER>>,
```

Comment:
Why?

Reply:
just like async_write, it needs to be used in other files  (such as read_index.rs) but under Peer struct.

Reply:
Then it can be accessed by `raft_group()` or `raft_group_mut()`.